Build a REST-API to add items in cart

chrill-christella opened the pull request:
### what does this PR do?

add items to buyers' cart

### Description of the task to be completed

. Add item to the buyer's cart  
.The product details (name, price, image) should be retrieved from the database  
.The updated cart should be stored in the buyer's account  
.A confirmation message should be sent to the frontend to be displayed to the buyer

### How should this be manually tested

. clone this repository  
. open it in your favorite editor  
.cd into it  
. checkout to ft-add-to-cart  
.run migrations and seeds by npm run migrate:up and npm run db:seeds respectively in your terminal  
. start the server by running npm run dev  
. make a post request in postman [http://localhost:${port}/api/cart/add/{productId}](http://localhost:${port}/api/cart/add/%7BproductId%7D) to add product into cart  
.You may also use the documented api by running your server and then in your browser http://localhost:${port}/docs
Relevant pivotal tracker user story
[#36]

oliviertech212 commented on src/controllers/cartController.js:
```diff
@@ -0,0 +1,61 @@
+      const {productId} = req.params;
+      const product = await Product.findOne({ where: { productId, available: true } });
+      if (!product) {
+        return res.status(404).json({ status: 404, message: "Product not found" });
```
you are duplicating statuscode
try
res.status(404).json({ status: "fail", message: "Product not found" });

oliviertech212 commented on src/controllers/cartController.js:
```diff
@@ -0,0 +1,61 @@
+      const {productId} = req.params;
+      const product = await Product.findOne({ where: { productId, available: true } });
+      if (!product) {
+        return res.status(404).json({ status: 404, message: "Product not found" });
```
you are duplicating status code.     check all
try
res.status(404).json({ status: "fail", message: "Product not found" });

chrill-christella replied:
yeah sure thank you!

sostene12 commented on src/controllers/cartController.js:
- you can change that status within json

sostene12 commented on test/buyer.test.js:
- also the tests are failing
![Uploading Screenshot 2023-04-04 at 10.37.12.png…]()

chrill-christella committed:
ft-add-to-cart:
implement cart.controller
implement cart migration
implement cart route and testing the endpoint
[finish]

chrill-christella commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Guest/Buyer should be able to add items to their cart](https://trello.com/c/LkfCZhS2/24-guest-buyer-should-be-able-to-add-items-to-their-cart)

wakaflorien reviewed (commented):
- make sure you resolve conflicts
